clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -909,6 +909,39 @@ public void ensureValid(String name, Object o) {
         }
     }
 
+    public static class ValidName implements Validator {
```

Comment:
I don't think these are general rules (i.e. do they apply to client IDs? Streams application IDs?). So perhaps move this to the ConnectorConfig class?